File: Keeper/Models/Keep.cs
using Keeper.Interfaces;
namespace Keeper.Models;


public class Keep : ICreated, IRepoItem<int>
{
  public int Id { get; set; }

  public Profile Creator { get; set; }
  public string CreatorId { get; set; }

  public string Img { get; set; }
  public string Description { get; set; }
  public string Name { get; set; }
  public int Views { get; set; }
  public int Kept { get; set; }

  public DateTime CreatedAt { get; set; }
  public DateTime UpdatedAt { get; set; }
}



public class KeepsInVault : Keep
{
  public int VaultKeepId { get; set; }
}

File: Keeper.client/src/models/Keep.js
export class Keep {
  constructor(data) {
    this.id = data.id
    this.vaultId = data.vaultId
    this.creatorId = data.creatorId
    this.vaultKeepId = data.vaultKeepId

    this.creator = data.creator

    this.createdAt = data.createdAt
    this.updatedAt = data.updatedAt

    this.img = data.img
    this.name = data.name
    this.description = data.description
    this.views = data.views
    this.kept = data.kept
  }
}